.happy/terraform/modules/sfn/main.tf

```diff
@@ -299,7 +299,16 @@ resource "aws_sfn_state_machine" "state_machine" {
             "MaxAttempts": 3,
             "BackoffRate": 2.0
         } ],
-        "Next": "DeregisterJobDefinition",
+        "Next": "DeregisterJobDefinitionAfterHandleErrors",
+        "ResultPath": null
+      },
+      "DeregisterJobDefinitionAfterHandleErrors": {
+        "Type": "Task",
+        "Next": "CheckForErrors",
```

Comment:
what if the downloader fails and there is not batch job registered?

Reply:
If the downloader fails and there is no task to deregister, then the deregistration will fail, which is fine; when an engineer comes to look at the step function, they'll see the Download failure.

Reply:
But I'm going to run a test scenario where Download fails...

Reply:
I _could_ add an empty default value in the input for $.batch.JobDefinitionName so that we send an empty value to `DeregisterJobDefinitionAfterHandleErrors`... but that would probably just cause the Deregister step to fail, which would mean it would still report as a failure albeit without going all the way to the end and getting to `DownloadValidateError`.

I don't think it's worth worrying about given the intention of these changes, which is to simply cause the overall sfn to fail, which it's doing. The cause is clear enough once you go to investigate, which you have to anyways since we're grouping Download errors with Validate errors.

Thoughts?

Reply:
As long as the step function is marked as failed, i don't mind.